Builds with make -j not working

Make check works, but make -j check does not.

Some dependency is not being tracked, which needs to be.

Not working again! 

```
make[1]: *** No rule to make target 'gptl.mod', needed by 'all-am'.  Stop.
make[1]: *** Waiting for unfinished jobs....
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -42,7 +42,7 @@ env:
 script:
   - autoreconf -i
   - ./configure
-  - make check
+  - make -j distcheck
 
 
  

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -19,6 +19,9 @@ lib_LTLIBRARIES = libgptlf.la
 libgptlf_la_SOURCES = gptlf.F90 printmpistatussize.F90	\
 process_namelist.F90
 
+# Each mod file depends on the .o file.
+gptl.mod: gptlf.$(OBJEXT)
+
 # Install these in the include directory.
 include_HEADERS = gptl.mod
 

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -22,3 +22,4 @@ process_namelist.F90
 # Install these in the include directory.
 include_HEADERS = gptl.mod
 
+CLEANFILES = *.mod

--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -61,7 +61,7 @@ threadohd_SOURCES = threadohd.F90 ${MODFILE}
 endif
 
 # Test output to be deleted.
-CLEANFILES = timing.*
+CLEANFILES = timing.* *.mod
 
 # This file is required for the nlreader test.
 EXTRA_DIST = gptlnl

